src/lib/cache/subCache.cpp

```diff
@@ -331,41 +337,38 @@ static bool attributeMatch(CachedSubscription* cSubP, const std::vector<std::str
 *
 * servicePathMatch -
 */
-static bool servicePathMatch(CachedSubscription* cSubP, char* servicePath)
+static bool servicePathMatch(CachedSubscription* cSubP, const std::string& servicePath)
 {
-  const char*  spath;
-
+  //
   // Special case. "/#" matches EVERYTHING
-  if ((strlen(servicePath) == 2) && (servicePath[0] == '/') && (servicePath[1] == '#') && (servicePath[2] == 0))
+  //
+  if (servicePath == "/#")
   {
     return true;
   }
 
   //
   // Empty service path?
   //
-  if ((strlen(servicePath) == 0) && (strlen(cSubP->servicePath) == 0))
+  if ((servicePath == "") && (cSubP->servicePath == ""))
```

Comment:
So, after looking to the anwers, it seems that the recommended way is not one nor the other :) but this:

```
if ((servicePath.empty()) && (cSubP->servicePath.empty()))
```